src/dataflow/render/mod.rs

```diff
@@ -591,6 +597,42 @@ pub(crate) fn build_dataflow<A: Allocate>(
     })
 }
 
+/// `arrange_from_upsert` may not behave as intended if multiple rows
```

Comment:
Moved this out of dataflow/decode since no decoding is actually taking place in this part of the code.